core/dbt/contracts/state.py

```diff
@@ -1,30 +1,48 @@
 from pathlib import Path
 from .graph.manifest import WritableManifest
 from .results import RunResultsArtifact
+from .results import FreshnessExecutionResultArtifact
 from typing import Optional
 from dbt.exceptions import IncompatibleSchemaException
 
 
 class PreviousState:
-    def __init__(self, path: Path):
+    def __init__(self, path: Path, current_path: Path):
```

Comment:
Two choices to get the unit tests passing:
1. Don't require `current_path` as an argument to initialize `PreviousState`, by setting the default arg value here to `None` and by adding an `if self.sources_current` conditional check below (line 34). This is the approach I prefer, as it's much simpler.
2. Every time we initialize `PreviousState` (including [`test/unit/test_graph_selector_methods.py:790`](https://github.com/dbt-labs/dbt-core/blob/5071b00baac3ff2a9bcce565d09661605de52654/test/unit/test_graph_selector_methods.py#L790)), pass in a placeholder value for `current_path`

Reply:
I went with choice 2 as `current_path` is best set within `runnable.py`. If not, we'll have to do lot of overhead implementations to avoid circular logic.